pxr/usd/plugin/usdDraco/exportAttribute.h

```diff
@@ -0,0 +1,229 @@
+//
+// Copyright 2018 Pixar
+//
+// Licensed under the Apache License, Version 2.0 (the "Apache License")
+// with the following modification; you may not use this file except in
+// compliance with the Apache License and the following modification to it:
+// Section 6. Trademarks. is deleted and replaced with:
+//
+// 6. Trademarks. This License does not grant permission to use the trade
+//    names, trademarks, service marks, or product names of the Licensor
+//    and its affiliates, except as required to comply with Section 4(c) of
+//    the License and to reproduce the content of the NOTICE file.
+//
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
+
+#ifndef USDDRACO_EXPORT_ATTRIBUTE_H
+#define USDDRACO_EXPORT_ATTRIBUTE_H
+
+#include "attributeDescriptor.h"
+
+#include "pxr/pxr.h"
+#include "pxr/usd/usdGeom/mesh.h"
+#include "pxr/usd/usdGeom/primvarsAPI.h"
+
+#include <draco/attributes/geometry_attribute.h>
+#include <draco/attributes/point_attribute.h>
+#include <draco/mesh/mesh.h>
+
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+
+/// \class UsdDracoExportAttribute
+///
+/// Helps to read and write mesh attributes while exporting USD files to Draco.
+///
+template <class ArrayT>
+class UsdDracoExportAttribute {
+public:
+    UsdDracoExportAttribute(UsdDracoAttributeDescriptor descriptor);
+    void GetFromMesh(const UsdGeomMesh &usdMesh, size_t numPositions);
```

Comment:
I don't quite understand what the functions GetFromMesh , GetFromRange, or SetToMesh are supposed to do (by looking at the function names alone)

Since they are part of the public interface, some comments would be probably good.

Reply:
Done. Added comments.